### DISCUSSION TITLE
Switch To Profile keeps asking to install plugin

### DISCUSSION THREAD

**Discussion by ForgetfulMemoryFoam** [elgatosf/elgato#118] [Help] [answered]
Hi everyone! I'm trying to implement an action that switches to another profile. 

So far, I've exported a profile named test. In my manifest.json:
```javascript
//manifest.json
	"Profiles": [
		{
			"Name": "profiles/test",
			"DeviceType": 0,
			"Readonly": false,
			"AutoInstall": true,
			"DontAutoSwitchWhenInstalled": false
		}
	]
```

With this set, I have inside of my app a function that is called if the action matches:
```javascript
const testing = (context, event, payload, device) => {

	if (event === 'keyDown') {
		console.log(payload["coordinates"])
		if (payload["coordinates"]["column"] === 0 && payload["coordinates"]["row"] === 2) {
			console.log('Switching away from test profile')
			// What about when we're not currently on the test profile?
			$SD.switchToProfile(device);
		} else {
			$SD.switchToProfile(device, 'test');
		}
	}
}
```

Even before I figure out how to get out of this profile (I know my above method won't work besides my one use case), my real issue is that every time I click on the new action key, it prompts to install this preconfigured plugin, even though my manifest is set to auto install. I install it and clicking on it goes into the new profile. When I return to the previous profile and click the action on the stream deck again, it prompts to install the plugin again. It does this continuously and keeps adding copies if I choose to install.

What am I missing here that it won't stop asking to install this plugin?

**Comment by ChekTek** [comment, marked as answer]
I believe the issue comes from calling `switchToProfile` without the directory included. Since the profile lives in the `profiles` directory, you'll want to include that in the request. 
```
$SD.switchToProfile(device, 'profiles/test');
```

Hope this helps!

**Reply by ForgetfulMemoryFoam** [reply]
I did that in my manifest and completely forgot in my app. Amazing, thank you for the quick response!
